src/QsCompiler/Tests.Compiler/GlobalVerificationTests.fs

```diff
@@ -34,19 +34,15 @@ type GlobalVerificationTests() =
         this.Expect "LocalNamespaceShortNames3" [ Error ErrorCode.UnknownType ]
         this.Expect "LocalNamespaceShortNames4" [ Error ErrorCode.UnknownType ]
         this.Expect "LocalNamespaceShortNames5" [ Error ErrorCode.UnknownIdentifier ]
-
-        this.Expect
-            "LocalNamespaceShortNames6"
-            [ Error ErrorCode.ArgumentTypeMismatch; Error ErrorCode.ArgumentTypeMismatch ]
-
+        this.Expect "LocalNamespaceShortNames6" [ Error ErrorCode.TypeMismatch; Error ErrorCode.TypeMismatch ]
         this.Expect "LocalNamespaceShortNames7" [ Error ErrorCode.UnknownIdentifier ]
         this.Expect "LocalNamespaceShortNames8" []
         this.Expect "LocalNamespaceShortNames9" []
         this.Expect "LocalNamespaceShortNames10" []
         this.Expect "LocalNamespaceShortNames11" [ Error ErrorCode.UnknownType; Error ErrorCode.UnknownIdentifier ]
         this.Expect "LocalNamespaceShortNames12" [ Error ErrorCode.UnknownIdentifier ]
         this.Expect "LocalNamespaceShortNames13" [ Error ErrorCode.UnknownType ]
-        this.Expect "LocalNamespaceShortNames14" [ Error ErrorCode.TypeMismatchInReturn ] // todo: could be more descriptive...
+        this.Expect "LocalNamespaceShortNames14" []
```

Comment:
The test case in question is

```qsharp
function LocalNamespaceShortNames14 () : (N4.IntPair -> Unit) {
    return N4.TakesAnyArg;
}
```

where

```qsharp
newtype IntPair = (N2.IntType, N3.IntType);
function TakesAnyArg<'T> (arg : 'T) : Unit {}
```

This code is valid since `TakesAnyArg.'T -> Unit` unifies with `IntPair -> Unit`, where `TakesAnyArg.'T` resolves to `IntPair`.